platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -4719,7 +4735,7 @@ void DisplayServerX11::process_events() {
 					// Ensure window focus on click.
 					// RevertToPointerRoot is used to make sure we don't lose all focus in case
 					// a subwindow and its parent are both destroyed.
-					if (!wd.no_focus && !wd.is_popup) {
+					if (!wd.no_focus && !wd.is_popup && _window_focus_check()) {
```

Comment:
This one probably should not be changed, since ButtonPress should always grab focus, the rest looks reasonable.

Reply:
Thanks! Updated in latest push